src/scripts/build_nst_da.py

```diff
@@ -0,0 +1,305 @@
+                zip.extractall(path=filename.replace(".zip", ""))
+            Path(filename).unlink()
+        case _:
+            pass
+
+
+def reorganise_files(dataset_dir: str | Path) -> None:
+    """Reorganise the files into `train` and `test` directories.
+
+    Args:
+        dataset_dir: The directory that should contain the dataset.
+    """
+    logger.info("Reorganising files")
+
+    data_dir = Path(dataset_dir)
+    train_dir = data_dir / "train"
+    test_dir = data_dir / "test"
+    train_audio_dir = train_dir / "audio"
+    test_audio_dir = test_dir / "audio"
+    train_audio_dir.mkdir(parents=True, exist_ok=True)
+    test_audio_dir.mkdir(parents=True, exist_ok=True)
+
+    for name in DATA_URLS:
+        if not Path(name).exists():
+            continue
+        match name:
+            case "train_metadata":
+                Path(name).rename("metadata")
+                shutil.move("metadata", train_dir)
+            case "train_audio":
+                data_dir = Path(name) / "dk"
+                for audio_dir in data_dir.iterdir():
+                    if not audio_dir.is_dir():
+                        continue
+                    for audio_file in audio_dir.glob("*.wav"):
+                        Path(audio_file).rename(train_audio_dir / audio_file.name)
+                shutil.rmtree(name)
+            case "test_and_errors":
+                data_dir = Path(name) / "supplement_dk"
+
+                temp_test_audio_dir = data_dir / "testdata" / "audio"
+                for audio_dir in temp_test_audio_dir.iterdir():
+                    if not audio_dir.is_dir():
+                        continue
+                    for audio_file in audio_dir.glob("*.wav"):
+                        Path(audio_file).rename(test_audio_dir / audio_file.name)
+                temp_test_metadata_dir = data_dir / "testdata" / "metadata"
+                shutil.move(temp_test_metadata_dir, test_dir)
+
+                test_log_file = data_dir / "testdata" / "sprakbanken_0611_transform.log"
+                Path(test_log_file).rename(data_dir / "test" / "log.log")
+
+                error_file = data_dir / "dk_errorfiles_train.json"
+                Path(error_file).rename(train_dir / "errorfiles.json")
+
+                test_manifest_file = data_dir / "dk_manifest_test.json"
+                Path(test_manifest_file).rename(test_dir / "manifest.json")
+
+                shutil.rmtree(name)
+            case "metadata_csvs":
+                train_metadata_csv = Path(name) / "NST_dk.csv"
+                Path(train_metadata_csv).rename(train_dir / "metadata.csv")
+                test_metadata_csv = Path(name) / "supplement_dk.csv"
+                Path(test_metadata_csv).rename(test_dir / "metadata.csv")
+                shutil.rmtree(name)
+            case "readme":
+                Path(name).rename("README.pdf")
+
+
+def remove_bad_files() -> None:
+    """Remove audio files that cannot be opened."""
+    bad_file_prefixes = [
```

Comment:
Are these found manually?

Reply:
Yep! I'll add that as a comment, as well as the function used to find them.